crates/sui-core/src/authority.rs

```diff
@@ -1590,100 +1592,173 @@ impl AuthorityState {
     }
 
     /// The object ID for gas can be any object ID, even for an uncreated object
+    #[allow(clippy::collapsible_else_if)]
     pub async fn dev_inspect_transaction_block(
         &self,
         sender: SuiAddress,
         transaction_kind: TransactionKind,
         gas_price: Option<u64>,
+        gas_budget: Option<u64>,
+        gas_sponsor: Option<SuiAddress>,
+        gas_objects: Option<Vec<ObjectRef>>,
+        epoch: Option<u64>,
+        skip_checks: Option<bool>,
     ) -> SuiResult<DevInspectResults> {
         let epoch_store = self.load_epoch_store_one_call_per_task();
+
         if !self.is_fullnode(&epoch_store) {
             return Err(SuiError::UnsupportedFeatureError {
                 error: "dev-inspect is only supported on fullnodes".to_string(),
             });
         }
 
+        if transaction_kind.is_system_tx() {
+            return Err(SuiError::UnsupportedFeatureError {
+                error: "system transactions are not supported".to_string(),
+            });
+        }
+        let skip_checks = skip_checks.unwrap_or(true);
+        let reference_gas_price = epoch_store.reference_gas_price();
         let protocol_config = epoch_store.protocol_config();
-        transaction_kind.check_version_supported(protocol_config)?;
-        transaction_kind.validity_check(protocol_config)?;
-
         let max_tx_gas = protocol_config.max_tx_gas();
-        let reference_gas_price = epoch_store.reference_gas_price();
-        let gas_price = match gas_price {
-            None => reference_gas_price,
-            Some(gas) => {
-                if gas == 0 {
-                    reference_gas_price
-                } else {
-                    gas
-                }
-            }
+
+        let gas_price = gas_price.unwrap_or(reference_gas_price);
+        let gas_budget = gas_budget.unwrap_or(max_tx_gas);
+        let owner = gas_sponsor.unwrap_or(sender);
+        // Payment might be empty here, but it's fine we'll have to deal with it later after reading all the input objects.
+        let payment = gas_objects.unwrap_or_default();
+        let expiration = match epoch {
+            None => TransactionExpiration::None,
+            Some(epoch) => TransactionExpiration::Epoch(epoch),
         };
```

Comment:
nit: this looks like a `map_or`:

```suggestion
        let expiration = epoch.map_or(TransactionExpiration::None, TransactionExpiration::Epoch);
```